Assert in `test_get_popular_torrents` (during the tests on Mac)

When testing #6600, I noticed the following stack trace, also see [here](https://jenkins-ci.tribler.org/job/GH_Tribler_PR_Tests/job/PR_mac_pytest/2722/testReport/junit/src.tribler-core.tribler_core.components.metadata_store.restapi.tests/test_channels_endpoint/test_get_popular_torrents/):

```
assert 29 == 30   +29   -30

add_fake_torrents_channels = None, mock_dlmgr_get_download = None
mock_dlmgr = <Mock id='4611317520'>
rest_api = <aiohttp.test_utils.TestClient object at 0x112dcebe0>

    async def test_get_popular_torrents(add_fake_torrents_channels, mock_dlmgr_get_download, mock_dlmgr, rest_api):
        """
        Test getting the list of popular torrents. The list is served as contents of a pseudo-channel
        """
        json_dict = await do_request(rest_api, 'channels/popular_torrents', expected_code=200)
    
        def fields(d, *args):
            return {key: d[key] for key in args}
    
        seeders_orig_order = [fields(d, 'type', 'num_seeders', 'num_leechers') for d in json_dict['results']]
    
        def sort_key(d):
            a = 1 if d["type"] == CHANNEL_TORRENT else 2 if d["type"] == COLLECTION_NODE else 3
            b = -d["num_seeders"]
            c = -d["num_leechers"]
            return (a, b, c)
    
        assert seeders_orig_order == sorted(seeders_orig_order, key=sort_key)
>       assert len(json_dict['results']) == 30  # torrents 1, 3, 5 in each of 10 channels
E       assert 29 == 30
E         +29
E         -30

src/tribler-core/tribler_core/components/metadata_store/restapi/tests/test_channels_endpoint.py:289: AssertionError
```

--- a/src/tribler/core/components/metadata_store/restapi/tests/conftest.py
+++ b/src/tribler/core/components/metadata_store/restapi/tests/conftest.py
@@ -1,3 +1,4 @@
+import hashlib
 from time import time
 
 from ipv8.keyvault.crypto import default_eccrypto
@@ -11,34 +12,44 @@ from tribler.core.components.metadata_store.utils import tag_torrent
 from tribler.core.utilities.utilities import random_infohash
 
 
+@db_session
+def create_channel(metadata_store, title, torrents_per_channel=5, local_version=0, subscribed=True):
+    def sha1_hash(value: str) -> bytes:
+        return hashlib.sha1(value.encode('utf-8')).digest()
+
+    def add_torrent_to_channel(torrent_title, seeders, self_checked):
+        t = metadata_store.TorrentMetadata(origin_id=channel.id_, title=torrent_title,
+                                           infohash=sha1_hash(torrent_title), sign_with=key)
+        t.health.seeders = seeders
+        t.health.self_checked = self_checked
+
+        one_week = 60 * 60 * 24 * 7
+        now = int(time())
+
+        t.health.last_check = now - one_week if self_checked else now
+
+    key = default_eccrypto.generate_key('curve25519')
+
+    channel = metadata_store.ChannelMetadata(title=title, subscribed=subscribed,
+                                             num_entries=torrents_per_channel,
+                                             infohash=sha1_hash(title), id_=123,
+                                             sign_with=key, version=10,
+                                             local_version=local_version)
+
+    for torrent_i in range(torrents_per_channel):
+        add_torrent_to_channel(f'torrent{channel.title}{torrent_i}', seeders=torrent_i,
+                               self_checked=bool(torrent_i % 2))
+
+
+@pytest.fixture
+def add_subscribed_and_not_downloaded_channel(metadata_store):
+    create_channel(metadata_store, 'Subscribed and not downloaded', subscribed=True, local_version=0)
+
+
 @pytest.fixture
 def add_fake_torrents_channels(metadata_store):
-    infohashes = []
-
-    torrents_per_channel = 5
-    # Add a few channels
-    with db_session:
-        for ind in range(10):
-            ext_key = default_eccrypto.generate_key('curve25519')
-            channel = metadata_store.ChannelMetadata(
-                title='channel%d' % ind,
-                subscribed=(ind % 2 == 0),
-                num_entries=torrents_per_channel,
-                infohash=random_infohash(),
-                id_=123,
-                sign_with=ext_key,
-                version=10,
-                local_version=(ind % 11),
-            )
-            for torrent_ind in range(torrents_per_channel):
-                rand_infohash = random_infohash()
-                infohashes.append(rand_infohash)
-                t = metadata_store.TorrentMetadata(
-                    origin_id=channel.id_, title='torrent%d' % torrent_ind, infohash=rand_infohash, sign_with=ext_key
-                )
-                t.health.seeders = int.from_bytes(t.infohash[:2], byteorder="big")
-                t.health.self_checked = bool(torrent_ind % 2 == 1)
-                t.health.last_check = int(time()) - (60 * 60 * 24 * 7 if torrent_ind % 2 else 0)
+    for i in range(10):
+        create_channel(metadata_store, f'channel{i}', subscribed=bool(i % 2), local_version=i)
 
 
 @pytest.fixture

--- a/src/tribler/core/components/metadata_store/restapi/tests/test_channels_endpoint.py
+++ b/src/tribler/core/components/metadata_store/restapi/tests/test_channels_endpoint.py
@@ -33,7 +33,8 @@ PNG_DATA = unhexlify(
     "fff03000006000557bfabd40000000049454e44ae426082"
 )
 
-# pylint: disable=unused-argument
+
+# pylint: disable=unused-argument, redefined-outer-name
 
 
 @pytest.fixture
@@ -58,15 +59,15 @@ def rest_api(loop, aiohttp_client, mock_dlmgr, metadata_store, tags_db):  # pyli
     return loop.run_until_complete(aiohttp_client(app))
 
 
-async def test_get_channels(rest_api, add_fake_torrents_channels, mock_dlmgr, metadata_store):
+async def test_get_channels(rest_api, add_fake_torrents_channels, add_subscribed_and_not_downloaded_channel, mock_dlmgr,
+                            metadata_store):
     """
     Test whether we can query some channels in the database with the REST API
     """
     mock_dlmgr.download_exists = lambda *args: None
     json_dict = await do_request(rest_api, 'channels')
-    assert len(json_dict['results']) == 10
-    # Default channel state should be METAINFO_LOOKUP
-    assert json_dict['results'][-1]['state'] == CHANNEL_STATE.METAINFO_LOOKUP.value
+    assert len(json_dict['results']) == 11
+    assert json_dict['results'][0]['state'] == CHANNEL_STATE.METAINFO_LOOKUP.value
 
     # We test out different combinations of channels' states and download progress
     # State UPDATING:
@@ -178,7 +179,7 @@ async def test_get_channel_contents_remote(metadata_store, add_fake_torrents_cha
 
 
 async def test_get_channel_contents_remote_request_timeout(
-    metadata_store, add_fake_torrents_channels, mock_dlmgr, rest_api
+        metadata_store, add_fake_torrents_channels, mock_dlmgr, rest_api
 ):
     """
     Test whether we can query torrents from a channel from a remote peer.
@@ -201,7 +202,7 @@ async def test_get_channel_contents_remote_request_timeout(
 
 
 async def test_get_channel_contents_remote_request_no_peers(
-    add_fake_torrents_channels, mock_dlmgr_get_download, rest_api, metadata_store
+        add_fake_torrents_channels, mock_dlmgr_get_download, rest_api, metadata_store
 ):
     """
     Test whether we can query torrents from a channel from a remote peer.
@@ -272,6 +273,8 @@ async def test_get_popular_torrents(add_fake_torrents_channels, mock_dlmgr_get_d
     Test getting the list of popular torrents. The list is served as contents of a pseudo-channel
     """
     json_dict = await do_request(rest_api, 'channels/popular_torrents', expected_code=200)
+    # torrents2 & torrent4 in each of 10 channels (but not torrent0, as it has 0 seeders)
+    assert len(json_dict['results']) == 20
 
     def fields(d, *args):
         return {key: d[key] for key in args}
@@ -285,7 +288,6 @@ async def test_get_popular_torrents(add_fake_torrents_channels, mock_dlmgr_get_d
         return (a, b, c)
 
     assert seeders_orig_order == sorted(seeders_orig_order, key=sort_key)
-    assert len(json_dict['results']) == 30  # torrents 1, 3, 5 in each of 10 channels
 
 
 async def test_get_popular_torrents_mdtype(add_fake_torrents_channels, mock_dlmgr_get_download, rest_api):
@@ -696,7 +698,8 @@ async def test_get_channel_thumbnail(rest_api, metadata_store):
             assert response.headers["Content-Type"] == "image/png"
 
 
-async def test_get_my_channel_tags(metadata_store, mock_dlmgr_get_download, my_channel, rest_api):  # pylint: disable=redefined-outer-name
+async def test_get_my_channel_tags(metadata_store, mock_dlmgr_get_download, my_channel,
+                                   rest_api):  # pylint: disable=redefined-outer-name
     """
     Test whether tags are correctly returned over the REST API
     """
@@ -713,7 +716,8 @@ async def test_get_my_channel_tags(metadata_store, mock_dlmgr_get_download, my_c
         assert len(item["tags"]) >= 2
 
 
-async def test_get_my_channel_tags_xxx(metadata_store, tags_db, mock_dlmgr_get_download, my_channel, rest_api):  # pylint: disable=redefined-outer-name
+async def test_get_my_channel_tags_xxx(metadata_store, tags_db, mock_dlmgr_get_download, my_channel,
+                                       rest_api):  # pylint: disable=redefined-outer-name
     """
     Test whether XXX tags are correctly filtered
     """

